### PR TITLE
Credit memo invoice allocations rework

### PR DESCRIPTION
**Pull request by Boydbueno**
## Summary

[Preview](https://preview.redoc.ly/rebilly-core/credit-memo-invoice-allocations-rework/)

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 1]
```diff
@@ -40,6 +40,35 @@ properties:
               $ref: ../CreatedTime.yaml
             updatedTime:
               $ref: ../UpdatedTime.yaml
+      invoices:
+        type: array
+        description: |-
+          List of invoices that the credit memo is allocated to.
```
Comment (noon-dawg) [inline openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: nitpick: multi-line is not needed here (I'm aware that I may have added this in the first place 😅)
```suggestion
        description: List of invoices that the credit memo is allocated to.
```

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 1 above)
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: Yup, I think you did :D But I like it, it looks good, don't you think so?

image

Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: We have the same format for transaction amount description, shown on the picture

Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: Also for readability on IDE, I think it's better to be on multi line as it's too long. I will leave as is if you don't mind

Comment (noon-dawg) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: yeah, that's cool. This is more of a note to myself 😄

**Review by noon-dawg — DISMISSED** [review]

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 1 above)
Comment (adamaltman) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:46]: No, this is a hard no for me @nanobash . If it's one line, don't use a block scalar.

**Review by adamaltman — CHANGES_REQUESTED** [review]

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 2]
```diff
@@ -40,6 +40,35 @@ properties:
...
+                - $ref: ../ResourceId.yaml
+            amount:
+              description: |-
+                Amount of credit that is allocated from the credit memo to the invoice.
+                If the `amount` value is not supplied or exceeds the unused amount of the credit memo or the invoice due amount, the lesser of the following two values is used:
+                - The unused amount of the credit memo.
+                - The invoice due amount.
+              type: number
+              format: double
+            currency:
```
[the remark this answers is not in the document]
Comment (adamaltman) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:62]: I think it should.

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 1 above)
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml]: Removed, thanks

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:61]: Yes, currencies are the same.

**Comment by nanobash** [comment]
@Rebilly/php please review

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 1 above)
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml]: Actually I mixed up at first, I meant amount descriptions, they have multi line descriptions with two `-` options.

**Comment by nanobash** [comment]
@Rebilly/php please review

**Comment by nanobash** [comment]
@Rebilly/php @Rebilly/docs please review

**Review by ashkarpetin — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Review by adamaltman — APPROVED** [review]

**Pull request merged**